win32/devdoc/job_object_helper.md

```diff
@@ -1,44 +1,123 @@
+MOCKABLE_FUNCTION(, THANDLE(JOB_OBJECT_HELPER), job_object_helper_create);
 ```
+``job_object_helper_create`` is used to create a `JOB_OBJECT_HELPER` object.
 
-`job_object_helper_limit_resources` is an API that uses the job object API to limit the amount of physical memory and CPU that the current process can consume.
+**SRS_JOB_OBJECT_HELPER_18_016: [** `job_object_helper_create` shall allocate a `JOB_OBJECT_HELPER` object. **]**
 
-**SRS_JOB_OBJECT_HELPER_18_001: [** If `percent_physical_memory` is `0`, `job_object_helper_limit_resources` shall fail and return a non-zero value. **]**
+**SRS_JOB_OBJECT_HELPER_18_017: [** `job_object_helper_create` shall call `sm_create`. **]**
 
-**SRS_JOB_OBJECT_HELPER_18_002: [** If `percent_cpu` is  `0`, `job_object_helper_limit_resources` shall fail and return a non-zero value. **]**
+**SRS_JOB_OBJECT_HELPER_18_018: [** If there are any failures, `job_object_helper_create` shall fail and return `NULL`. **]**
 
-**SRS_JOB_OBJECT_HELPER_18_004: [** If `percent_physical_memory` is greater than `100`, `job_object_helper_limit_resources` shall fail and return a non-zero value. **]**
+**SRS_JOB_OBJECT_HELPER_18_019: [** `job_object_helper_create` shall succeed and return the `JOB_OBJECT_HELPER` object.  **]**
 
-**SRS_JOB_OBJECT_HELPER_18_005: [** If `percent_cpu` is greater than `100`, `job_object_helper_limit_resources` shall fail and return a non-zero value. **]**
 
-**SRS_JOB_OBJECT_HELPER_18_006: [** `job_object_helper_limit_resources` shall call `CreateJobObject` to create a new job object passing `NULL` for both `lpJobAttributes` and `lpName`. **]**
+## job_object_helper_dispose
+```c
+static void job_object_helper_dispose(JOB_OBJECT_HELPER* dispose);
+```
+`job_object_helper_dispose` frees all of the resources used by the `JOB_OBJECT_HELPER` object.
+
+**SRS_JOB_OBJECT_HELPER_18_020: [** `job_object_helper_dispose` shall call `sm_destroy`. **]**
+
+
+## job_object_helper_open
+```c
+MOCKABLE_FUNCTION(, int, job_object_helper_open, THANDLE(JOB_OBJECT_HELPER), job_object_helper);
+```
+`job_object_helper_open` prepares the `JOB_OBJECT_HELPER` object for use by initializing the resources that it needs. 
+
+**SRS_JOB_OBJECT_HELPER_18_021: [** If `job_object_helper` is `NULL`, `job_object_helper_open` shall fail and return a non-zero value. **]**
+
+**SRS_JOB_OBJECT_HELPER_18_022: [** `job_object_helper_open` shall call `sm_open_begin`. **]**
+
+**SRS_JOB_OBJECT_HELPER_18_023: [** `job_object_helper_open` shall call `GlobalMemoryStatusEx` to get the total amount of physical memory in kb. **]**
 
-**SRS_JOB_OBJECT_HELPER_18_007: [** `job_object_helper_limit_resources` shall call `GetCurrentProcess` to get the current process handle. **]**
+**SRS_JOB_OBJECT_HELPER_18_024: [** `job_object_helper_open` shall call `CreateJobObject` to create a new job object passing `NULL` for both `lpJobAttributes` and `lpName`. **]**

```

Comment:
>CreateJobObject [](http://example.com/codeflow?start=73&length=15)

this for sure cannot be in the "open", needs to be moved to "create". 

I also suspect open and close and not needed (together with SM).